web/api/v0/__init__.py

```diff
@@ -326,6 +328,85 @@ def get(self, ipv4: IPv4Address | IPv6Address) -> ResponseReturnValue:
 
 api.add_resource(UserByIPResource, '/user/from-ip')
 
+class MPSKSClientAddResource(Resource):
+    @use_kwargs(
+        {
+            "password": fields.Str(required=True),
+            "mac": fields.Str(required=True),
+            "name": fields.Str(required=True),
+        },
+        location="form",
+    )
+    def post(self, user_id: int, password: str, mac: str, name: str) -> ResponseReturnValue:
+        user = get_authenticated_user(user_id, password)
+
+        try:
+            mpsk_client_create(user, mac, name, user)
+            session.session.commit()
+        except InvalidMACAddressException:
+            abort(400, message="Invalid MAC address.")
+        except IntegrityError:
+            abort(400, message="Mac address is already in use.")
+        return "mpsk has been added."
```

Comment:
I think it should not be too hard to test the API: we already have frontend tests. So all you need to do is add an `api` package to `tests.frontend` and execute tests in a similar way (all you should need are some fixtures to get a filled database and a `client`).

See https://github.com/flask-restful/flask-restful/pull/110/files for an example of how to use `url_for`, which needs to happen a bit differently because we don't specify endpoint names.

Reply:
You might have to set up an API token to pass it, but that should be the only nontrivial thing.